homeassistant/components/deluge/__init__.py

```diff
@@ -1 +1,356 @@
+                )
+            )
+
+        def add_torrent(service):
+            """Add new torrent to download."""
+            torrent = service.data[ATTR_TORRENT]
+            deluge_client = None
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == service.data[CONF_HOST]:
+                    deluge_client = self.hass.data[DOMAIN][entry.entry_id]
+                    break
+            if deluge_client is None:
+                _LOGGER.error("Deluge host is not found")
+                return
+            if torrent.startswith(("http", "ftp:")):
+                deluge_api.core.add_torrent_url(torrent, {})
+            elif torrent.startswith(("magnet:")):
+                deluge_api.core.add_torrent_magnet(torrent, {})
+            else:
+                _LOGGER.warning("Could not add torrent: unsupported type")
+
+        self.hass.services.async_register(
+            DOMAIN, SERVICE_ADD_TORRENT, add_torrent, schema=SERVICE_ADD_TORRENT_SCHEMA
+        )
+
+        self.config_entry.add_update_listener(self.async_options_updated)
+
+        return True
+
+    def add_options(self):
+        """Add options for entry."""
+        if not self.config_entry.options:
+            scan_interval = self.config_entry.data.pop(
+                CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+            )
+            options = {CONF_SCAN_INTERVAL: scan_interval}
+
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, options=options
+            )
+
+    def set_scan_interval(self, scan_interval):
+        """Update scan interval."""
+
+        def refresh(event_time):
+            """Get the latest data from Transmission."""
+            self._deluge_data.update()
+
+        if self.unsub_timer is not None:
+            self.unsub_timer()
+        self.unsub_timer = async_track_time_interval(
+            self.hass, refresh, timedelta(seconds=scan_interval)
+        )
+
+    @staticmethod
+    async def async_options_updated(hass, entry):
+        """Triggered by config entry options updates."""
+        hass.data[DOMAIN][entry.entry_id].set_scan_interval(
+            entry.options[CONF_SCAN_INTERVAL]
+        )
+
+
+class DelugeData:
+    """Get the latest data and update the states."""
+
+    def __init__(self, hass, config, api):
+        """Initialize the Deluge Data object."""
+        self.hass = hass
+        self.config = config
+        self.data = None
+        self.torrents = None
+        self.session = None
+        self.available = True
+        self._api = api
+        self.completed_torrents = []
+        self.started_torrents = []
+
+    @property
+    def host(self):
+        """Return the host name."""
+        return self.config.data[CONF_HOST]
+
+    @property
+    def signal_update(self):
+        """Update signal per deluge entry."""
+        return f"{DATA_UPDATED}-{self.host}"
```

Comment:
Why use host and not config entry id ?

Reply:
Sure we can use entry id. I will fix it.